config/config.go

```diff
@@ -181,8 +181,7 @@ type AgentConfig struct {
 	// TODO(cam): Remove UTC and parameter, they are no longer
 	// valid for the agent config. Leaving them here for now for backwards-
 	// compatibility
-	// Deprecated: 1.0.0 after, has no effect
-	UTC bool `toml:"utc"`
+	UTC bool `toml:"utc" deprecated:"1.0.0;option is ignored"`
```

Comment:
@reimda, 

@MyaLongmire and I are going through these and we think the changes look good, but the concern is around the message of: `1.0.0;option is ignored`

Per the spec we wrote, I think we were hoping for a message that directed the user to action like:

```
this option was ignored as of 1.0.0
```

Another example:


```
1.7.0;use 'tls_ca' instead
```

to

```
ssl_ca was deprecated in 1.7.0, use 'tls_ca' instead
```

Thoughts?

Reply:
@jpowers I think the deprecated struct tag expects a value in the format "version;reason" and later on we have a line that adds the boilerplate like `printf("%s was deprecated in %s, %s", settingName, version, reason)`

I agree the "option is ignored" text should be more descriptive but I think we can come up with something descriptive that fits in the boilerplate.

If we use the struct tag `deprecated:"1.0.0,the agent autodetects whether the timezone is UTC so the setting is ignored"` we'll get something in the user friendly format we expect:

```
utc was deprecated in 1.0.0, the agent autodetects whether the timezone is UTC so the setting is ignored
```

Reply:
@reimda  exactly. @powersj the deprecation framework parses the struct tags and convert them to a "nice" (YMMV) message shown to the user, so you just specify `<version of deprecation>[;version of removal];<notice shown to the user>` in the tag.

Reply:
Thanks for that.

@Hipska - Thanks for cleaning all of these up. I think the next step is to clean up the notices.

Reply:
Which notices?